### ISSUE TITLE
Indentations in the middle

### ISSUE THREAD

**Issue by edemaine** [DanielXMoore/Civet#27]
CoffeeScript allows for indentation in the middle of e.g. an array literal:

```coffee
[1
 2
 3]
```

Civet won't parse this, even after #26.

More generally, I think whenever we have indented forms, we often need to support switching from regular form to indented form, at least once. For example:
```coffee
[1, 2, 3   # regular form
 4, 5, 6   # switch to indented
 7, 8, 9]
```

Oddly CoffeeScript (see `ArgElisionList` in `grammar.coffee`) supports additional indentation blocks in the middle of an array literal:
```coffee
[1
 2
   3
]
```
This is probably less important to support. But one indentation level is pretty common I think.

**Comment by STRd6** [comment]
This would be nice to have. I think a place to start would be in `ArrayLiteralContent` adjust the rule to have an inline element list followed by `NestedElementList`.
